padrino-gen/lib/padrino-gen/padrino-tasks/activerecord.rb

```diff
@@ -402,6 +416,36 @@ def less_than_active_record_5_2?
     ActiveRecord.version < Gem::Version.create("5.2.0")
   end
 
+  def less_than_active_record_6_0?
+    ActiveRecord.version < Gem::Version.create("6.0.0")
+  end
+
+  def with_database(env_name)
+    if less_than_active_record_6_0?
+      config = ActiveRecord::Base.configurations.with_indifferent_access[env_name]
+
+      yield config
+    else
+      db_configs = ActiveRecord::Base.configurations.configs_for(env_name: env_name.to_s)
+
+      db_configs.each do |db_config|
+        yield db_config.config
```

Comment:
ar:setup fails here. Needs to be:

```ruby
config = db_config.config.with_indifferent_access
yield config
```ruby

Reply:
@basex Hmm... My app is no problem

```ruby
# config/database.rb

ActiveRecord::Base.configurations = YAML.load(<<~YAML)
development:
  adapter: sqlite3
  database: db/app_development.db
YAML
```

```bash
$ bundle exec rake ar:setup
  DEBUG -   (1.5ms)  SELECT sqlite_version(*)
 already exists
-- create_table("companies", {:force=>:cascade})
  DEBUG -   (0.2ms)  DROP TABLE IF EXISTS "companies"
  DEBUG -   (3.4ms)  CREATE TABLE "companies" ("id" integer PRIMARY KEY AUTOINCREMENT NOT NULL, "no" integer NOT NULL, "company_name" varchar NOT NULL, "receipted_date" varchar NOT NULL, "destination_name" text, "created_at" datetime, "updated_at" datetime)
  DEBUG -   (2.8ms)  CREATE UNIQUE INDEX "index_companies_on_company_name_and_receipted_date" ON "companies" ("company_name", "receipted_date")
   -> 0.0082s
  DEBUG -   (5.8ms)  CREATE TABLE "schema_migrations" ("version" varchar NOT NULL PRIMARY KEY)
  DEBUG -   (0.2ms)  SELECT "schema_migrations"."version" FROM "schema_migrations" ORDER BY "schema_migrations"."version" ASC
  DEBUG -   (2.6ms)  INSERT INTO "schema_migrations" (version) VALUES (20141119131622)
  DEBUG -   (3.6ms)  INSERT INTO "schema_migrations" (version) VALUES
(20141103085818);


  DEBUG -   (3.8ms)  CREATE TABLE "ar_internal_metadata" ("key" varchar NOT NULL PRIMARY KEY, "value" varchar, "created_at" datetime(6) NOT NULL, "updated_at" datetime(6) NOT NULL)
  DEBUG -  ActiveRecord::InternalMetadata Load (0.2ms)  SELECT "ar_internal_metadata".* FROM "ar_internal_metadata" WHERE "ar_internal_metadata"."key" = ? LIMIT ?  [["key", "environment"], ["LIMIT", 1]]
  DEBUG -   (0.1ms)  begin transaction
  DEBUG -  ActiveRecord::InternalMetadata Create (0.3ms)  INSERT INTO "ar_internal_metadata" ("key", "value", "created_at", "updated_at") VALUES (?, ?, ?, ?)  [["key", "environment"], ["value", "development"], ["created_at", "2019-10-07 17:13:02.917220"], ["updated_at", "2019-10-07 17:13:02.917220"]]
  DEBUG -   (2.6ms)  commit transaction
```

Reply:
You don't see the problem because you are using sqlite.

[L62](https://github.com/padrino/padrino-framework/blob/5444214cb8a46387128c0732ad4562954e570c9e/padrino-gen/lib/padrino-gen/padrino-tasks/activerecord.rb#L62) to [L86](https://github.com/padrino/padrino-framework/blob/5444214cb8a46387128c0732ad4562954e570c9e/padrino-gen/lib/padrino-gen/padrino-tasks/activerecord.rb#L86) doesn't apply to you. This logic depends on the keys being symbols, but they are strings without the change, so nothing on the switch case runs.

Reply:
@basex I see. I added at 5e4f75c